### ISSUE TITLE
Add GitHub issues as tasks

### ISSUE THREAD

**Issue by Dyljyn** [ParabolInc/action#2879]
## Issue - Enhancement

I had my first meeting using this tool today and the biggest feedback I got from the team, is the disconnect with our already existing GitHub setup. 
- It felt repetitive to write a task for an issue that already exists in our GitHub.
- No reference to an issue in which more details are provided.

The existing integration of adding a task as an issue in GitHub is fine for those who use the tool from the beginning but not for those who already have issues in GitHub or when these issues are created by others (for example, users with access to the GitHub repository).

### Acceptance Criteria (optional)
Users can:
   - Import individual issues from GitHub whenever it is possible to create a task.
   - Link a task to a GitHub issue

**Estimated effort:** X points ([see CONTRIBUTING.md](https://github.com/ParabolInc/action/blob/master/CONTRIBUTING.md#points-and-sizes))

**Comment by ackernaut** [comment]
Hey @Dyljyn glad to see you raise this issue! Thank you for clearly describing the UX pains you’re having with out GitHub integration.

### Re: import existing issues

I’m curious about importing GitHub issues. I see you mentioned being able to import individual issues. Does this need happen mostly in a meeting or on the team dashboard? Would you ever want to import many issues at once?

### Re: link a task to an issue

You can format content in a task card, including adding a link. This is a hidden affordance to keep the UI minimal. Just press `command + k` (or the equivalent for Windows) on selected text to add a link (like Google Docs, if you use that). To see other formatting tips, just press `?` on a view that has cards, but not while editing a card. Does that get you want you need in regards to links?

Feel free to write in `love@parabol.co` and let us know how Parabol is working for your team, and how we can improve.

Thanks, Terry

**Comment by Dyljyn** [comment]
Linking is indeed possible that way and sufficient enough.

---

To elaborate on the importing of GitHub: We would essentially like to have this tool work with our GitHub issues instead of "creating new tasks that link to our GitHub issues".

Internally we use GitHub's project boards for managing releases, which works great but are having trouble with defining sprints. We tried using GitHub's milestones but it requires quite the effort from our team. Furthermore, the meetings weren't as structured as I intend them to be sometimes. This is where this tool came into the picture.

To answer the question of multiple issues. I wouldn't say it's a high demand to add multiple issues but rather that it would save some time, instead of: `click (+), click "import GitHub issue", find issue, import issue.`

**TL;DR**
- Mostly during meetings
- Yes to multiple issues but low priority

**Comment by ackernaut** [comment]
Very helpful notes @Dyljyn ! Our intention has always been to improve the GitHub functionality over time, but per our roadmap strategy, we prioritize features where the UX pains are based on real user feedback, so it’s very helpful to hear about our GitHub integration. Let’s keep in touch!

**Comment by jordanh** [comment]
In the future (where everything is better) we hope to extend task cards so that they can be linked easily to existing issues, probably by implementing a search function right on the card face itself. Thank you for your feedback and I agree with @ackernaut—let's keep in touch!

**Comment by jordanh** [comment]
This is still a thing we want to do – likely by adding an index of integrated objects to Parabol and making it easy to reference/link to these objects via Task cards. It's not an immediate priority but will very likely be revisited over 2020.

Issue came up during scrub. Iceboxing for now.
